packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -17,79 +17,508 @@ import {
+
+export type SelectProps<T> = Omit<
+  React.ComponentPropsWithoutRef<'div'>,
+  'onChange' | 'placeholder' | 'value' | 'defaultValue'
+> &
+  (
+    | ({
+        /**
+         * If true, the native `<select>` element will be rendered.
+         *
+         * Extra props, such as `name` can be passed to the `<select>` using `triggerProps`.
+         *
+         * @default false
+         */
+        native: true;
+      } & NativeSelectProps)
+    | ({ native?: false } & CustomSelectProps<T>)
+  );
+
+// ----------------------------------------------------------------------------
+
+const NativeSelect = React.forwardRef((props, forwardedRef) => {
+  const {
+    triggerProps,
+    options,
+    disabled,
+    placeholder,
+    defaultValue: defaultValueProp = placeholder !== undefined ? '' : undefined,
+    value: valueProp,
+    onChange: onChangeProp,
+    size,
+    status,
+    required,
+    ...rest
+  } = props;
+
+  return (
+    <InputWithIcon {...rest} ref={forwardedRef}>
+      <SelectButton
+        as='select'
+        size={size}
+        status={status}
+        disabled={disabled}
+        defaultValue={valueProp === undefined ? defaultValueProp : undefined}
+        value={valueProp === null ? '' : valueProp}
+        required={required}
+        {...triggerProps}
+        onKeyDown={mergeEventHandlers(triggerProps?.onKeyDown, (event) => {
+          // Firefox does not open the menu on Enter, so we need to do it manually.
+          if (event.key === 'Enter') {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            event.currentTarget.showPicker?.();
+          }
+        })}
+        onChange={mergeEventHandlers(triggerProps?.onChange, (event) => {
+          onChangeProp?.(event.currentTarget.value, event);
+        })}
+      >
+        {placeholder !== undefined ? (
+          <option value='' disabled>
+            {placeholder}
+          </option>
+        ) : null}
+
+        {options.map((option) => (
+          <option key={option.value} {...option}>
+            {option.label}
+          </option>
+        ))}
+      </SelectButton>
+
+      <SelectEndIcon disabled={disabled} />
+    </InputWithIcon>
+  );
+}) as PolymorphicForwardRefComponent<'div', NativeSelectProps>;
+
+// ----------------------------------------------------------------------------
+
+// This is a smaller, stricter subset of CustomSelectProps, because native
+// <select> only supports string values, and does not support icons or sublabels.
+type NativeSelectProps = SelectCommonProps & {
   /**
-   * Close handler that closes the dropdown.
+   * Selected option value.
+   *
+   * Must be a string, because it is passed as an attribute to the native <select>.
+   *
+   * Alternatively, pass `null` to reset the value.
    */
-  close: () => void;
+  value?: string | null;
   /**
-   * Indicates whether an item is selected.
+   * Callback invoked when the selected value changes.
```

Comment:
When `native` is not passed, looks like both the JSDocs sentences are being shown.

<img width="637" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/409e58f8-5404-4944-96f0-efc3b1ac56bf">

<img width="629" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/f8afdffa-6b4f-4d07-88f0-331d490d5622">

So maybe the part where `SelectProps` changes based `native` needs to be modified a bit.

Reply:
I think JSDocs are just broken for `CustomSelectProps`. If you hover a prop, it doesn't even appear (same as https://github.com/iTwin/iTwinUI/pull/1862#discussion_r1492672122). 

We should try to simplify some of the types and figure out what is going on (in the future). I don't want to make big changes to the types in this PR.

Reply:
Sounds good, then maybe we can add this point to the after-PR TODOs.